### PR TITLE
feat: add image optimization using api

### PR DESCRIPTION
**Pull request by 0xabdulkhalid**
## Description
- This PR is about optimizing the github user's profile image using image optimizing api
- This api is used to convert `png` image to `webp` during load time, an `png` image may size more than `150kb` but after conversion to `webp` the same image may size more than `40kb`
- We don't need to fear about the failure of api, because it can cache the image for more than 2 days so once an image the next it will immediately serve the process imaged within seconds
- This is way more important to increase the site performance by decreasing the consuming bandwidth
- This PR closes the ISSUE: #304 

## Pull Request Checklist

- [x] The app is working correctly after applying the changes.
- [x] The app is responsive and displays properly on different screen sizes and devices.
- [x] The changes are thoroughly documented in the appropriate files or documentation sections.
- [x] The coding conventions and style guidelines are followed consistently.

## Checklist

- [x] I have tested the changes locally and ensured that the app is functioning as expected.
- [x] I have verified that the app is responsive and looks good on various screen sizes and devices.
- [x] I have reviewed the documentation and ensured that it accurately reflects the changes made.
- [x] I have followed the coding conventions and style guidelines defined in the project.
- [x] I have run the automated tests and all of them pass.
- [x] I have considered the performance implications of the changes made.
- [x] I have considered the security implications of the changes made.
- [x] I have checked for any potential regressions or side effects caused by the changes.
- [x] I have performed a self-review of my code.
- [x] Starred the repository.
- [x] Added My Name to the Contributor's List.
- [x] Followed the repository's [Contributing Guidelines](/CONTRIBUTING.md).
- [x] I ran the app and tested it locally to verify that it works as expected.

## Reviewers
 
@tusharbansal22 , @AmanJain18

### CODE REVIEW

**Review by anmode — APPROVED** [review]
Curious to know. what does it optimize? In terms of size or pixels? fetch from github

**Comment by 0xabdulkhalid** [comment]
@anmode , The image from the Github API is served in `PNG` format which it can sizes about `150kb` and greater but our idea is to optimize the image by converting it during the load itself using an Third party API

The converted `webp` only sizes lesser than `50kb`, the optimized image is cached on api's server once requested so the next time we request it can serve the image without any further process.

The Image Optimization API is well suited for this scenarios, can convert any images into `webp`, this is a next-generation format introduced by Google.

Nowdays we want to optimize the resources in every aspect to improve the site's performance 🚀

**Comment by anmode** [comment]
Ohh okay got that. Thanks  @0xabdulkhalid

**Comment by 0xabdulkhalid** [comment]
As per request of @AmanJain18 I have edited on another stuff on this same issue.

This edit is about fixing the improper transition of Profile card image during 'hover'

For some reason i can't add the video footage of before and after

**Comment by AmanJain18** [comment]
Ya i got it but still it can't be merged...

**Comment by 0xabdulkhalid** [comment]
@AmanJain18 Can i know why ?

File: styles/Contributor.module.css [hunk 1]
```diff
@@ -5,14 +5,14 @@
   border-radius: 50%;
   height: 200px;
   margin-bottom: 20px;
+  transition: .4s ease;
```
Comment (AmanJain18) [inline styles/Contributor.module.css:8]: make this 0.3s ease-in-out

File: styles/Contributor.module.css [hunk 2]
```diff
@@ -5,14 +5,14 @@
...
   @media screen and (max-width: 768px) {
     height: 150px;
   }
 }
 .profile:hover {
    box-shadow: 0 0 11px rgba(66, 44, 255, 0.8), 0 0 11px rgba(102, 204, 255, 0.8);
   transform: scale(1.1);
-  transition: 0.3s ease-in-out;
```
Comment (AmanJain18) [inline styles/Contributor.module.css:15]: don't remove this line and align box shadow to transform

**Review by AmanJain18 — CHANGES_REQUESTED** [review]
Make that changes

**Review by AmanJain18 — APPROVED** [review]

**Pull request merged**